core/src/main/scala/org/apache/spark/api/python/SerDeUtil.scala

```diff
@@ -106,8 +106,15 @@ private[spark] object SerDeUtil extends Logging {
    * Convert an RDD of Java objects to an RDD of serialized Python objects, that is usable by
    * PySpark.
    */
-  def javaToPython(jRDD: JavaRDD[_]): JavaRDD[Array[Byte]] = {
-    jRDD.rdd.mapPartitions { iter => new AutoBatchedPickler(iter) }
+  def javaToPython(jRDD: JavaRDD[_], batchSize: Integer): JavaRDD[Array[Byte]] = {
```

Comment:
we can make `batchSize` an optional parameter so call site doesn't need to change

Reply:
Good point!